From a6bf8d5c754dcde068dedd2e826ae95af50c197e Mon Sep 17 00:00:00 2001
From: Ralf Wildenhues <Ralf.Wildenhues@gmx.de>
Date: Tue, 24 Nov 2009 11:36:53 +0100
Subject: [PATCH 1/2] Fix AC_FUNC_MMAP regression with C++ compiler in 2.65.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

* lib/autoconf/functions.m4 (AC_FUNC_MMAP): Use const char*
for the constant string.  Cast void* to char* for assignment.
* NEWS, THANKS: Update.
Report by Michal Čihař.

Signed-off-by: Ralf Wildenhues <Ralf.Wildenhues@gmx.de>
---
 ChangeLog                 |    8 ++++++++
 NEWS                      |    2 ++
 THANKS                    |    1 +
 lib/autoconf/functions.m4 |    7 ++++---
 4 files changed, 15 insertions(+), 3 deletions(-)

diff --git ChangeLog ChangeLog
index d8d0f9b..521ef79 100644
--- ChangeLog
+++ ChangeLog
@@ -1,3 +1,11 @@
+2009-11-24  Ralf Wildenhues  <Ralf.Wildenhues@gmx.de>
+
+	Fix AC_FUNC_MMAP regression with C++ compiler in 2.65.
+	* lib/autoconf/functions.m4 (AC_FUNC_MMAP): Use const char*
+	for the constant string.  Cast void* to char* for assignment.
+	* NEWS, THANKS: Update.
+	Report by Michal Čihař.
+
 2009-11-21  Eric Blake  <ebb9@byu.net>
 
 	Release Version 2.65.
diff --git NEWS NEWS
index b72eb17..d05b78c 100644
--- NEWS
+++ NEWS
@@ -1,5 +1,7 @@
 GNU Autoconf NEWS - User visible changes.
 
+** AC_FUNC_MMAP works in C++ mode again.  Regression introduced in 2.65.
+
 * Major changes in Autoconf 2.65 (2009-11-21) [stable]
   Released by Eric Blake, based on git versions 2.64.*.
 
diff --git THANKS THANKS
index b288163..fdd6930 100644
--- THANKS
+++ THANKS
@@ -250,6 +250,7 @@ Matthew D. Langston         langston@SLAC.Stanford.EDU
 Matthew Mueller             donut@azstarnet.com
 Matthew Woehlke             mw_triad@users.sourceforge.net
 Matthias Andree             matthias.andree@gmx.de
+Michal Čihař                nijel@debian.org
 Michael Elizabeth Chastain  chastain@cygnus.com
 Michael Jenning             ?
 Michael Matz                matz@kde.org
diff --git lib/autoconf/functions.m4 lib/autoconf/functions.m4
index 6b6e7fc..14a8cb9 100644
--- lib/autoconf/functions.m4
+++ lib/autoconf/functions.m4
@@ -1258,6 +1258,7 @@ int
 main ()
 {
   char *data, *data2, *data3;
+  const char *cdata2;
   int i, pagesize;
   int fd, fd2;
 
@@ -1282,10 +1283,10 @@ main ()
   fd2 = open ("conftest.txt", O_RDWR | O_CREAT | O_TRUNC, 0600);
   if (fd2 < 0)
     return 4;
-  data2 = "";
-  if (write (fd2, data2, 1) != 1)
+  cdata2 = "";
+  if (write (fd2, cdata2, 1) != 1)
     return 5;
-  data2 = mmap (0, pagesize, PROT_READ | PROT_WRITE, MAP_SHARED, fd2, 0L);
+  data2 = (char *) mmap (0, pagesize, PROT_READ | PROT_WRITE, MAP_SHARED, fd2, 0L);
   if (data2 == MAP_FAILED)
     return 6;
   for (i = 0; i < pagesize; ++i)
-- 
1.6.5.3

